### PR TITLE
[Interop] [SwiftToCxx] Introduce Dynamic Cast to swift::Error

### PR DESCRIPTION
**Pull request by Robertorosmaninho**
This modification allow the user to cast a generic `swift::Error` to its own Error and identify which case was catch!

### CODE REVIEW

**Comment by hyp** [comment]
@swift-ci please test

File: lib/PrintAsClang/_SwiftCxxInteroperability.h [hunk 1]
```diff
@@ -191,6 +195,46 @@ extern "C" void *_Nonnull swift_errorRetain(void *_Nonnull swiftError) noexcept;
...
+                                  uint32_t flags);
+
+struct SymbolicP {
+  alignas(2) uint8_t _1;
+  uint32_t _2;
+  uint8_t _3[2];
+  uint8_t _4;
+} __attribute__((packed));
+
+inline const void *_Nullable testErrorCall() {
```
Comment (hyp) [inline lib/PrintAsClang/_SwiftCxxInteroperability.h:220]: let's rename it to `getErrorMetadata` or something like that.

**Comment by hyp** [comment]
@swift-ci please test

File: lib/PrintAsClang/_SwiftCxxInteroperability.h [hunk 2]
```diff
@@ -21,9 +21,13 @@
 
 #include <cstdint>
 #include <stdlib.h>
+#include <cstdint>
```
Comment (hyp) [inline lib/PrintAsClang/_SwiftCxxInteroperability.h:24]: redundant - there's an include for it two lines above.

File: test/Interop/SwiftToCxx/functions/swift-functions-errors-execution.cpp [hunk 3]
```diff
@@ -1,8 +1,8 @@
 // RUN: %empty-directory(%t)
 
-// RUN: %target-swift-frontend %S/swift-functions-errors.swift -typecheck -module-name Functions -clang-header-expose-decls=all-public -emit-clang-header-path %t/functions.h
+// RUN: %target-swift-frontend %S/swift-functions-errors.swift -typecheck -module-name Functions -clang-header-expose-decls=has-expose-attr -emit-clang-header-path %t/functions.h
 
-// RUN: %target-interop-build-clangxx -c %s -I %t -o %t/swift-functions-errors-execution.o
+// RUN: %target-interop-build-clangxx -std=c++17 -x objective-c++ -c %s -I %t -o %t/swift-functions-errors-execution.o
```
Comment (hyp) [inline test/Interop/SwiftToCxx/functions/swift-functions-errors-execution.cpp:5]: Do you still need the Objective-C++ mode here, or can we use C++?

Comment (Robertorosmaninho) [reply test/Interop/SwiftToCxx/functions/swift-functions-errors-execution.cpp:5]: No, we don't 😄 
I'll delete it!

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Comment by hyp** [comment]
@swift-ci please test

**Comment by hyp** [comment]
@swift-ci please test source compatibility

**Pull request merged**